File: nats/python/publisher.py
#!/usr/bin/env python3.12

# Simple NATS Python Publisher
# Requires: pip install nats-py
import asyncio
from nats.aio.client import Client as NATS
import sys

async def main():
    if len(sys.argv) < 4:
        print(f"Usage: {sys.argv[0]} <server> <subject> <message>")
        return
    server = sys.argv[1]
    subject = sys.argv[2]
    message = sys.argv[3]

    nc = NATS()
    await nc.connect(servers=[server])
    await nc.publish(subject, message.encode())
    print(f"Published to {subject}: {message}")
    await nc.drain()

if __name__ == "__main__":
    asyncio.run(main())

File: nats/cpp/publisher.cpp
// Simple NATS C++ Publisher Example
// Requires: https://github.com/nats-io/nats.c
#include <nats/nats.h>
#include <iostream>

int main(int argc, char* argv[]) {
    if (argc < 4) {
        std::cout << "Usage: " << argv[0] << " <server> <subject> <message>" << std::endl;
        return 1;
    }
    const char* server = argv[1];
    const char* subject = argv[2];
    const char* message = argv[3];

    natsConnection* conn = nullptr;
    natsStatus s = natsConnection_ConnectTo(&conn, server);
    if (s != NATS_OK) {
        std::cerr << "Failed to connect: " << natsStatus_GetText(s) << std::endl;
        return 1;
    }
    s = natsConnection_PublishString(conn, subject, message);
    if (s != NATS_OK) {
        std::cerr << "Failed to publish: " << natsStatus_GetText(s) << std::endl;
        natsConnection_Destroy(conn);
        return 1;
    }
    std::cout << "Published to " << subject << ": " << message << std::endl;
    natsConnection_Destroy(conn);
    return 0;
}
